src/core/methods/createChannel.ts

```diff
@@ -3,134 +3,72 @@
 import {
   ChannelState,
   FeedType,
-  MembraneType,
-  FluxExpressionReference,
 } from "@/store/types";
-import { v4 } from "uuid";
-import { Perspective, Link } from "@perspect3vism/ad4m";
 import type { PerspectiveHandle } from "@perspect3vism/ad4m";
-import { createNeighbourhoodMeta } from "./createNeighbourhoodMeta";
-import { SOCIAL_CONTEXT_OFFICIAL } from "@/constants/languages";
 import { ad4mClient } from "@/app";
 import { CHANNEL } from "@/constants/neighbourhoodMeta";
+import { nanoid } from "nanoid";
 
 interface ChannelProps {
   channelName: string;
   creatorDid: string;
   sourcePerspective: PerspectiveHandle;
-  membraneType: MembraneType;
-  typedExpressionLanguages: FluxExpressionReference[];
 }
 
 export async function createChannel({
   channelName,
   creatorDid,
   sourcePerspective,
-  membraneType,
-  typedExpressionLanguages,
 }: ChannelProps): Promise<ChannelState> {
-  const perspective = await ad4mClient.perspective.add(channelName);
-  console.debug("Created new perspective with result", perspective);
-  const socialContextLanguage =
-    await ad4mClient.languages.applyTemplateAndPublish(
-      SOCIAL_CONTEXT_OFFICIAL,
-      JSON.stringify({
-        uid: v4().toString(),
-        name: `${channelName}-social-context`,
-      })
-    );
-  console.debug(
-    "Created new social context language wuth result",
-    socialContextLanguage
-  );
-
-  //Publish perspective
-  const metaLinks = await createNeighbourhoodMeta(
-    channelName,
-    "",
-    creatorDid,
-    typedExpressionLanguages
-  );
-
-  const meta = new Perspective(metaLinks);
-
-  const neighbourhood = await ad4mClient.neighbourhood.publishFromPerspective(
-    perspective.uuid,
-    socialContextLanguage.address,
-    meta
-  );
-  console.debug("Create a neighbourhood with result", neighbourhood);
+  const id = nanoid();
 
   const addLinkToChannel = await ad4mClient.perspective.addLink(
     sourcePerspective.uuid,
     {
       source: sourcePerspective.sharedUrl!,
-      target: neighbourhood,
+      target: id,
       predicate: CHANNEL,
     }
   );
+  
   console.debug(
     "Created new link on source social-context with result",
     addLinkToChannel
   );
 
-  //Add link on channel social context declaring type
-  const addChannelTypeLink = await ad4mClient.perspective.addLink(
-    perspective.uuid,
+  const channelNameLink = await ad4mClient.perspective.addLink(
+    sourcePerspective.uuid,
     {
-      source: neighbourhood,
-      target: CHANNEL,
-      predicate: "rdf://type",
+      source: addLinkToChannel.data.target,
+      target: channelName,
+      predicate: 'flux://name',
```

Comment:
If we have the channel name as the id, then this can be removed and instead we would add a link which looks like:

```
{
    source: "channelName",
    predicate: "ad4m://has_class",
    target: "flux://channel"
}
```